vm_dump.c

```diff
@@ -994,36 +997,36 @@ rb_vm_bugreport(const void *ctx)
 #ifdef __APPLE__
 	vm_address_t addr = 0;
 	vm_size_t size = 0;
-	struct vm_region_submap_info_64 map;
-	mach_msg_type_number_t count = VM_REGION_SUBMAP_INFO_COUNT_64;
+	struct vm_region_submap_info map;
+	mach_msg_type_number_t count = VM_REGION_SUBMAP_INFO_COUNT;
 	natural_t depth = 0;
 
 	fprintf(stderr, "* Process memory map:\n\n");
 	while (1) {
-		if (vm_region_recurse_64(mach_task_self(), &addr, &size, &depth,
-			(vm_region_recurse_info_t)&map, &count) != KERN_SUCCESS) {
-			break;
-		}
-
-		if (map.is_submap) {
-			// We only look at main addresses
-			depth++;
-		} else {
-			fprintf(stderr, "%lu-%lu %s%s%s", addr, (addr+size),
-				((map.protection & VM_PROT_READ) != 0 ? "r" : "-"),
-				((map.protection & VM_PROT_WRITE) != 0 ? "w" : "-"),
-				((map.protection & VM_PROT_EXECUTE) != 0 ? "x" : "-"));
+        if (vm_region_recurse(mach_task_self(), &addr, &size, &depth,
```

Comment:
Why removed the indents?
You may want to indent by spaces?

Reply:
you are right sorry.